nima/webserver/webserver/src/main/java/io/helidon/nima/webserver/http/HttpRouting.java

```diff
@@ -278,60 +278,54 @@ public Builder any(String pattern, Handler handler) {
         }
     }
 
-    private static class RoutingHandler implements Handler {
+    private static final class RouteExecutor implements Runnable {
         private final ConnectionContext ctx;
         private final RoutingRequest request;
         private final RoutingResponse response;
         private final ServiceRoute rootRoute;
 
-        private RoutingHandler(ServiceRoute rootRoute, ConnectionContext ctx, RoutingRequest request, RoutingResponse response) {
+        private RouteExecutor(ServiceRoute rootRoute, ConnectionContext ctx, RoutingRequest request, RoutingResponse response) {
             this.rootRoute = rootRoute;
             this.ctx = ctx;
             this.request = request;
             this.response = response;
         }
 
         @Override
-        public void handle(ServerRequest req, ServerResponse res) {
+        public void run() {
             try {
-                doHandle();
-            } catch (HttpException e) {
-                ctx.simpleHandlers().handle(e, response);
-            }
-        }
+                HttpPrologue prologue = request.prologue();
+                response.resetRouting();
 
-        private void doHandle() {
-            // ignore parameters, as we need instances of routing request (and we do not want to do an explicit cast)
-            HttpPrologue prologue = request.prologue();
-            response.resetRouting();
-
-            RoutingResult result = RoutingResult.ROUTE;
-            int counter = 0;
-            while (result == RoutingResult.ROUTE) {
-                counter++;
-                if (counter == 10) {
-                    LOGGER.log(System.Logger.Level.ERROR, "Rerouted more than 10 times. Will not attempt further routing");
-
-                    throw HttpException.builder()
-                            .request(HttpSimpleRequest.create(prologue, request.headers()))
-                            .type(SimpleHandler.EventType.INTERNAL_ERROR)
-                            .build();
-                }
+                RoutingResult result = RoutingResult.ROUTE;
+                int counter = 0;
+                while (result == RoutingResult.ROUTE) {
+                    counter++;
+                    if (counter == 10) {
```

Comment:
Magic number could be configurable

Reply:
Yes it could :) I will add it to the task list (not part of this PR)